### DISCUSSION TITLE
DNS requests from within build hosts?

### DISCUSSION THREAD

**Discussion by timwoj** [cirruslabs/cirrus-ci-docs#958] [Q&A] [answered]
I recently added some tests to one of our components that make DNS requests and check for valid results (it's a DNS resolver library). I noticed that any of the tests that make `PTR` requests were failing, so I investigated via one of the FreeBSD VMs. Lo and behold, any `PTR` request I make from that VM (using `dig`) doesn't return any results. `A` and `TXT` requests are fine, but `PTR` requests fail. I didn't test any of the other VMs, but a bunch of them failed (see https://cirrus-ci.com/build/6262895343304704).

Is there reason for that? Should I open a bug?

**Comment by fkorotkov** [comment, marked as answer]
I can do `dig -x 1.1.1.1` in a regular `container`-based task. We do have some firewall rules which blocks some ports and some traffic patterns but the network settings are the same for containers and FreeBSD VMs. I've also tried to reproduce the issue on a FreeBSD VM with the following config:

```yaml
drill_task:
  freebsd_instance:
    image_family: freebsd-13-0
  script: 
    - drill -x 1.1.1.1
```

But my task succeeded https://cirrus-ci.com/task/5968131502899200?logs=main#L0. Maybe there is something specific about your tests?

**Reply by timwoj** [reply]
Looks like it was actually something specific about the DNS library we were using previously. I was working on adding a bunch of tests prior to switching out the library for something else, and the new one works fine. Sorry for the noise.
